vertx-opentracing/src/test/java/io/vertx/tracing/opentracing/it/BasicTracingIT.java

```diff
@@ -0,0 +1,99 @@
+package io.vertx.tracing.opentracing.it;
+
+import java.time.Duration;
+
+import io.opentracing.Tracer;
+import io.restassured.parsing.Parser;
+import io.restassured.response.Response;
+import io.vertx.core.Vertx;
+import io.vertx.core.VertxOptions;
+import io.vertx.junit5.Checkpoint;
+import io.vertx.junit5.VertxExtension;
+import io.vertx.junit5.VertxTestContext;
+import io.vertx.tracing.opentracing.OpenTracingOptions;
+import io.vertx.tracing.opentracing.it.jaegercontainer.JaegerContainerAllIn;
+import org.junit.jupiter.api.AfterAll;
+import org.junit.jupiter.api.BeforeAll;
+import org.junit.jupiter.api.DisplayName;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.extension.ExtendWith;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import static io.restassured.RestAssured.given;
+import static org.awaitility.Awaitility.await;
+import static org.hamcrest.Matchers.hasItems;
+import static org.hamcrest.Matchers.hasSize;
+
+@ExtendWith(VertxExtension.class)
+public class BasicTracingIT {
+  private static final Logger LOGGER = LoggerFactory.getLogger(BasicTracingIT.class);
+
+  private static Vertx tracedVertx;
+
+  private static Tracer tracer;
+
+  private static final String JAEGER_SERVICE_NAME = "test-traced-service";
+
+  private static final JaegerContainerAllIn JAEGER_ALL_IN_ONE = new JaegerContainerAllIn("quay.io/jaegertracing/all-in-one:latest");
+
+  @BeforeAll
+  public static void deploy(VertxTestContext context) {
+    Checkpoint deployCheck = context.checkpoint(2);
+    context.verify(() -> {
+      JAEGER_ALL_IN_ONE.start();
+      tracer = JAEGER_ALL_IN_ONE.createTracer(JAEGER_SERVICE_NAME);
+      deployCheck.flag();
+
+      tracedVertx = Vertx.vertx(
+        new VertxOptions().setTracingOptions(
+          new OpenTracingOptions(tracer))
+      );
+
+      tracedVertx.deployVerticle(ServerVerticle.class.getName(), context.succeeding(id -> {
+        await("Await until route is ready!").until(() -> {
```

Comment:
It's not a good practice to use blocking libraries such as `awaitility` on a vert.x event loop. You should either use Vert.x jUnit 5 tools or block in the main test thread (with the help of a `CountDownLatch`).

Reply:
Ok I see it, I will use vert.x junit5 tools, thank you @tsegismont !